### PR TITLE
Vodafone SMS Notification

### PR DESCRIPTION
**Pull request by Human018**
Use Vodafone (Australia) SMS API to send Laravel Notifications

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,109 @@
-# New Notification Channels
+# SMS Notifications with Vodafone
```
Comment (atymic) [inline README.md]:
```suggestion
# Vodafone Notification Channel
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,109 @@
...
+[![Software License](https://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat-square)](LICENSE.md)
+[![Build Status](https://img.shields.io/travis/laravel-notification-channels/vodafone/master.svg?style=flat-square)](https://travis-ci.org/laravel-notification-channels/vodafone)
+[![Quality Score](https://img.shields.io/scrutinizer/g/laravel-notification-channels/vodafone.svg?style=flat-square)](https://scrutinizer-ci.com/g/laravel-notification-channels/vodafone)
+[![Code Coverage](https://img.shields.io/scrutinizer/coverage/g/laravel-notification-channels/vodafone/master.svg?style=flat-square)](https://scrutinizer-ci.com/g/laravel-notification-channels/vodafone/?branch=master)
+[![Total Downloads](https://img.shields.io/packagist/dt/laravel-notification-channels/vodafone.svg?style=flat-square)](https://packagist.org/packages/laravel-notification-channels/vodafone)
 
-### I'm working on a new channel
-Please create an issue for it if it does not already exist, then PR you code for review.
+This package makes it easy to send sms notifications using [Vodafone](https://www.vodafone.com.au/messaging/smser) with Laravel 5.5+ and 6.0
```
Comment (atymic) [inline README.md]:
```suggestion
This package makes it easy to send sms notifications using [Vodafone](https://www.vodafone.com.au/messaging/smser) with Laravel 5.5+ and 6.x
```

File: src/VodafoneChannel.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+use Illuminate\Notifications\Notification;
+use NotificationChannels\Vodafone\Exceptions\CouldNotSendNotification;
+
+class VodafoneChannel
+{
+    /**
+     * @var string Vodafone's API endpoint
+     */
+    protected $endpoint = 'https://www.smsertech.com/apisend';
```
Comment (atymic) [inline src/VodafoneChannel.php]: Do you have a copy or link to the API docs?

File: src/VodafoneChannel.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     * Send the given notification.
+     *
+     * @param mixed $notifiable
+     * @param \Illuminate\Notifications\Notification $notification
+     *
+     * @throws \NotificationChannels\Vodafone\Exceptions\CouldNotSendNotification
+     */
+    public function send($notifiable, Notification $notification)
+    {
+        $message = $notification->toVodafone($notifiable, $notification);
```
Comment (atymic) [inline src/VodafoneChannel.php]: I'd double check before this that the method exists, and then after to make sure it returned a valid object.

File: src/VodafoneChannel.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        $fields = [
```
Comment (atymic) [inline src/VodafoneChannel.php]: Can we please use guzzle or another modern HTTP client here.

for example with guzzle the block below is just:

```
try {
	(new Client())->post(['form_params' => [/* unencoded data here */]);
} catch($e) {
	// handle error
}

Comment (atymic) [reply src/VodafoneChannel.php]: I'd also prefer it if this was split out into a separate client class and bound in the service container.

File: src/VodafoneChannel.php (hunk 3 above)
Comment (Human018) [reply src/VodafoneChannel.php]: Yes I do, Vodafone only provides a PDF of the documentation, which I have now included in the package files.

Comment (atymic) [reply src/VodafoneChannel.php]: Thanks :)

File: src/VodafoneChannel.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        /* Confirm the toVodafone method exists before continuing */
+        if (! method_exists($notification, 'toVodafone')) {
+            throw CouldNotSendNotification::methodDoesNotExist();
+        }
+
+        $message = $notification->toVodafone($notifiable, $notification);
+
+        /* Check notification uses correct class for this API */
+        if (! $message instanceof VodafoneMessage) {
```
Comment (atymic) [inline src/VodafoneChannel.php]: One thing i would do here is accept a string and then just create a new `VodafoneMessage` :)

**Comment by atymic** [comment]
Repo is here. Please do the final PR there and then i'll finish the checklist :)
https://github.com/laravel-notification-channels/vodafone

https://github.com/laravel-notification-channels/vodafone/issues/1

**Pull request closed without merging**